packages/tpetra/core/src/Tpetra_Details_FixedHashTable_def.hpp

```diff
@@ -1142,6 +1142,15 @@ init (const keys_type& keys,
     Kokkos::deep_copy (counts, countsHost);
   }
 
+  // FENCE REVIEW - NOW PASSING
+  //   Testing: This code is exercised by unit tests.
+  //   GTX960:  Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+  //   White:   Passed with CUDA_LAUNCH_BLOCKING=0 and fence removed.
+  //   Plan:    Remove this fence if we refactor code below to avoid UVM access.
+  //   Notes:   Need to check debug path as well.
+  //            Seems the passes are false oks - this code could fail.
+  //            For example see Mark's comments: 'This assumes UVM'
+
```

Comment:
@kddevin The code has two routes. One is parallel build of the hash table. The other is serial build from host. So, fence is only required when it uses parallel build. Otherwise, mirroring would apply fence inside when it uses UVM. 

@crtrott To make it sure, does deep copy on UVM apply fence inside ?

Reply:
On Kokkos' slack, @crtrott says "yes it does if you don't hand it a space argument"

Reply:
We need to leave a note that fence is required when 3 argument deep copy is used in the above. It looks like that we are changing deep copy from 2 argument interface to 3 argument interface. 

Also, it is a bit odd and could be problematic if deep copy behaves differently with uvm and without uvm for the 3 argument deep copy.

Reply:
Updated comments #7546